paymentsheet/src/main/java/com/stripe/android/common/analytics/experiment/LogLinkGlobalHoldbackExposure.kt

```diff
@@ -1,35 +1,112 @@
+    @IOContext private val workContext: CoroutineContext,
+    private val customerRepository: CustomerRepository,
     private val logger: Logger
-) {
+) : LogLinkGlobalHoldbackExposure {
 
-    operator fun invoke(
+    override operator fun invoke(
         elementsSession: ElementsSession,
-    ) = runCatching {
+        state: PaymentElementLoader.State
+    ) {
         if (FeatureFlags.linkGlobalHoldbackExposureEnabled.isEnabled) {
-            val experimentsData = requireNotNull(
-                elementsSession.experimentsData
-            ) { "Experiments data required to log exposures" }
-            val holdbackOn = elementsSession.linkSettings?.linkGlobalHoldbackOn == true
-            eventReporter.onExperimentExposure(
-                experiment = LinkGlobalHoldback(
-                    arbId = experimentsData.arbId,
-                    group = if (holdbackOn) TREATMENT else CONTROL,
-                ),
-            )
+            CoroutineScope(workContext).launch {
+                runCatching {
+                    logExposure(elementsSession, state)
+                }.onFailure { error ->
+                    logger.error("Failed to log Global holdback exposure", error)
+                }
+            }
         }
-    }.onFailure { error ->
-        logger.error("Failed to log Global holdback exposure", error)
     }
+
+    private suspend fun logExposure(
+        elementsSession: ElementsSession,
+        state: PaymentElementLoader.State
+    ) {
+        val experimentsData = requireNotNull(
+            elementsSession.experimentsData
+        ) { "Experiments data required to log exposures" }
+        val experimentGroup: String = elementsSession.experimentsData
+            ?.experimentAssignments[LINK_GLOBAL_HOLD_BACK] ?: "control"
+
+        val customerEmail = state.getEmail()
+
+        val isReturningUser: Boolean = customerEmail?.let { isReturningUser(email = it) } == true
+
+        logger.debug(
+            """Link Global Holdback exposure: 
+                |arbId=${experimentsData.arbId},
+                |isReturningLinkConsumer=$isReturningUser,
+                |group=$experimentGroup
+            """.trimMargin()
+        )
+        eventReporter.onExperimentExposure(
+            experiment = LinkGlobalHoldback(
+                arbId = experimentsData.arbId,
+                isReturningLinkConsumer = isReturningUser,
+                group = experimentGroup,
+            ),
+        )
+    }
+
+    suspend fun isReturningUser(
+        email: String,
+    ): Boolean {
+        return linkDisabledApiRepository
+            .lookupConsumerWithoutBackendLoggingForExposure(email)
+            .map { it.exists }
+            .fold(
+                onSuccess = { it },
+                onFailure = {
+                    logger.error("Failed to check if user is returning", it)
+                    false
+                }
+            )
```

Comment:
Nit: No need to `fold`.

```suggestion
            .map { it.exists }
            .onFailure {
                logger.error("Failed to check if user is returning", it)
                false
            }
```

Reply:
onFailure does not allow error recovery, but `getOrElse` does the trick!